include/path.hpp

```diff
@@ -0,0 +1,141 @@
+#pragma once
+
+#include <iostream>
+#include <vector>
+#include <eigen3/Eigen/Dense>
+#include <eigen3/Eigen/Core>
+#include <glm/glm.hpp>
+
+
+// -------------------------------------------------------------------------------------------------------------------//
+
+// Misc definitions
+
+
+// Flags for boundary conditions
+// Since Eigen arrays don't work well with enum types, we just take integer flags
+typedef int bc_t;
+static const bc_t BC_T_START = 0, BC_ZERO = 0, BC_NATURAL = 1, BC_CLAMP = 2, BC_T_END = 3;
```

Comment:
The C++ way is:
```cpp
enum class boundary_condition  : int { 
    start,
    zero,
    natural,
    clamp,
    end
};
```
if you want to force their values for whatever reason, you can set them explicitly. This has the advantage that these things are scoped without ugly C-style namespacing. You use them as:
```cpp
auto bc = boundary_condition::natural;
```